src/Views/DeviceView.vala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2011–2021 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA.
+ */
+
+public class Power.DeviceView : Granite.SimpleSettingsPage {
+
+    private Services.Device battery;
+    private string device_path = "";
+
+    public DeviceView (
+        string path,
+        Gtk.Widget icon,
+        string status_device,
+        string title,
+        bool show_header
+    ) {
+        //  this.device_path = path; // TODO: FIX 
```

Comment:
@cassidyjames How can I initialize the device_path here ? 
This line are throwing ```segmentation fault error```